js-extensions/src/ClassMetadataStore.js

```diff
@@ -8,13 +8,20 @@ export class ClassMetadataStore {
          * Type info cache
          */
         this.classMetadata = {};
-        
+
         /**
          * Fetch function for the classMetadata
          */
         this.classMetadataProvider = classMetadataProvider;
+
+        /**
+         * Onload callbacks cache. Used to ensure we don't
+         * issue multiple in-parallel requests for the same
+         * class metadata.
+         */
+        this.classMetadataOnloadCallbacks = {};
```

Comment:
Perhaps we could create a ticket to track the work? Otherwise it's just tech debt we can't easily measure.